includes/wikia/services/WAMService.class.php

```diff
@@ -89,6 +89,40 @@ public function getCurrentWamScoreForWiki ($wikiId) {
 	}
 
 	/**
+	 * Returns the latest WAM rank provided a wiki ID
+	 * @param int $wikiId
+	 * @return number
+	 */
+	public function getCurrentWamRankForWiki( $wikiId ) {
+		$memKey = wfSharedMemcKey( 'datamart', self::MEMCACHE_VER, 'wam_rank', $wikiId );
+
+		$getData = function () use ( $wikiId ) {
+			if ( $this->isDisabled() ) {
+				return 0;
+			}
+
+			$db = $this->getDB();
+
+			$result = $db->selectField(
+				[ 'fact_wam_scores' ],
+				'wam_rank',
+				[ 'wiki_id' => $wikiId ],
+				__METHOD__,
+				[
+					'ORDER BY' => 'time_id DESC',
+					'LIMIT' => 1
+				]
+			);
+
+			return $result ? $result : 0;
+		};
+
+		$wamRank = WikiaDataAccess::cacheWithLock( $memKey, self::CACHE_DURATION, $getData );
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$wamRank". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)